### PR TITLE
remove duping the response array, uunecessary overhead for pagination

### PR DESCRIPTION
**Pull request by Mandaryn**

### CODE REVIEW

File: lib/bookingsync/api/client.rb [hunk 1]
```diff
@@ -209,7 +209,8 @@ def paginate(path, options = {}, &block)
           response = call(request_method, path, options)
         end
         @pagination_first_response = response
-        data = response.resources.dup
+        data = []
+        data.concat(response.resources)
```
Comment (Fredar) [inline lib/bookingsync/api/client.rb]: Why do we need this change?
This code is executed only once, right?

Comment (Mandaryn) [inline lib/bookingsync/api/client.rb]:
```
[18] pry(main)> Benchmark.bm(27) do |bm|
[18] pry(main)*   bm.report('dup') do  
[18] pry(main)*     iterations.times do |n|    
[18] pry(main)*       array = ["the", {"key" => "responses#{n}"}, "from", "api", "v3#{n}"].dup      
[18] pry(main)*       array.concat(["more", {"key" => "responses#{n}"}, "from", "api", "v3#{n}"])      
[18] pry(main)*     end      
[18] pry(main)*   end    
[18] pry(main)*   
[18] pry(main)*   bm.report('nodup') do  
[18] pry(main)*     iterations.times do |n|    
[18] pry(main)*       array = []      
[18] pry(main)*       array.concat(["the", {"key" => ["responses#{n}"]}, "from", "api", "v3#{n}"])      
[18] pry(main)*       array.concat(["more", {"key" => ["responses#{n}"]}, "from", "api", "v3#{n}"])      
[18] pry(main)*     end      
[18] pry(main)*   end    
[18] pry(main)* end  
                                  user     system      total        real
dup                           0.150000   0.160000   0.310000 (  0.359492)
nodup                         0.090000   0.030000   0.120000 (  0.120827)
```

Comment (Mandaryn) [inline lib/bookingsync/api/client.rb]: this is for just small content and almost 3x faster, duping the response hash is unnecessary overhead

Comment (Fredar) [inline lib/bookingsync/api/client.rb]: Ok I have run it for 100 million iterations :smile: 
```
                                  user     system      total        real
dup                          33.540000 323.650000 357.190000 (359.187487)
nodup                        30.850000 312.870000 343.720000 (344.817036)
```
Its not ground-breaking, but looks a bit faster, lets do it.

Comment (Mandaryn) [inline lib/bookingsync/api/client.rb]: 4% faster, small steps

**Pull request merged**